### PR TITLE
Automate hash calculation; update sources to alma 8.9

### PR DESCRIPTION
**Pull request by h-vetinari**
Makes it easier to update to newer alma versions (including minor ones). For major bumps, this will even modify the subfolders of the RPMs accordingly.

~Builds on #77 & #78, so review those first.~

### CODE REVIEW

**Comment by h-vetinari** [comment]
With the automation here, the alma 9 update in #80 only requires the following changes (aside from unavoidable adaptations to infrastructure changes in alma itself):
```diff
diff --git a/recipe/conda_build_config.yaml b/recipe/conda_build_config.yaml
index 0c256de..97f1b44 100644
--- a/recipe/conda_build_config.yaml
+++ b/recipe/conda_build_config.yaml
@@ -15,22 +15,22 @@ cross_target_platform:
   - linux-s390x

 alma_version:
-  - "8.9"
+  - "9.3"
 glibc_version:
-  - "2.28"
+  - "2.34"
 kh_version:
-  - "4.18.0"
+  - "5.14.0"
 # glibc artefacts have two build numbers plus the alma version, e.g.
 #   2.28-236.el8_9.13.x86_64.rpm
 #   ↑    ↑     ↑   ↑
 #   └glibc_ver └alma_version
 #        └build1   └build2
 glibc_build1:
-  - "236"
+  - "83"
 glibc_build2:
-  - "13"
+  - "12"
 kh_build:
-  - "513.24.1"
+  - "362.24.2"
 # nss-* versions need to be set in meta.yaml (and/or update.py)

 ctng_vendor:
diff --git a/recipe/update.py b/recipe/update.py
index 8f816de..edbc25b 100644
--- a/recipe/update.py
+++ b/recipe/update.py
@@ -54,7 +54,7 @@ name2string = {
     "nss_db": glibc_string,
     # manual override
     "nss_nis": "3.0-8.el8",
-    "nss-softokn-freebl": "3.90.0-6.el8_9",
+    "nss-softokn-freebl": "3.90.0-7.el9_3",
 }

 def get_subfolder(pkg, string):
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -102,15 +112,15 @@ build:
 
 outputs:
   - name: kernel-headers_{{ cross_target_platform }}
-    version: {{ kernel_headers_version }}
+    version: {{ kh_version }}
```
Comment (isuruf) [inline recipe/meta.yaml:115]: Let's keep the name here

Comment (h-vetinari) [reply recipe/meta.yaml:115]: It just gets very long if we use that prefix consistently... 🤷

**Comment by h-vetinari** [comment]
FWIW, the changes here are much easier to review commit by commit

**Review by beckermr — CHANGES_REQUESTED** [review]
This came up on the core call. What are the ABI/API implications for updating the kernel headers? I have a similar question for any other version updates as well. 

cc @conda-forge/core

**Comment by h-vetinari** [comment]
> This came up on the core call.

Which one?

> What are the ABI/API implications for updating the kernel headers?

The kernel headers are not being updated here, they stay on 4.18.0 just as they were before:
https://github.com/conda-forge/linux-sysroot-feedstock/blob/9a37264a65458167ad33132ca498acdc9c73a017/recipe/meta.yaml#L2

The entire alma 8.x line is ABI-compatible (which is the main selling point), so going from 8.7 to 8.9 should have absolutely no impact. Going to alma 9 (not in this PR, but #80) is a different story, but also not any different from how we're already handling alma8 vs cos7 (and previously vs. cos60.

**Review by beckermr — APPROVED** [review]

**Comment by isuruf** [comment]
Thanks

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

File: .ci_support/linux_64_centos_machineaarch64cross_target_platformlinux-aarch64target_machineaarch64.yaml [hunk 2]
```diff
@@ -12,6 +12,8 @@ ctng_vendor:
 - conda
 docker_image:
 - quay.io/condaforge/linux-anvil-cos7-x86_64
+glib:
+- '2'
```
Comment (isuruf) [inline .ci_support/linux_64_centos_machineaarch64cross_target_platformlinux-aarch64target_machineaarch64.yaml:16]: @beckermr, see https://github.com/conda/conda-build/pull/5447/files/70d396c8e535ea1e9b4696f2513e2fa13d542470#diff-ab3e639bcffb453109c8b994f783028c3fbfdf8dc16655bdfd1849385738bf42

**Pull request merged**